lib/eblox/data/provider.ex

```diff
@@ -53,33 +58,34 @@ defmodule Eblox.Data.Provider do
   def perform(:died, _id, payload), do: {:transition, :*, payload}
   def perform(_state, _id, payload), do: {:transition, :scan, payload}
 
-  @spec handle_changes(t()) :: [:ok]
-  def handle_changes(%Provider{created: _, deleted: _, changed: _} = changes) do
+  @spec handle_changes(module(), t()) :: [:ok]
```

Comment:
This is an extremely important part of writing the idiomatic _Elixir_ code. In line 75 below you expect this module to respond to `initial_payload/1`. Yes, we cannot define a type that would reflect a behaviour implementation, but we at least can help future us to understand what’s going on here.

So, please declare a behaviour for `impl` in the first place. Then declare `@type t :: module()` there and use `ImplBeh.t()` here in typespec, so that readers of the code know where to look for hints/docs.